src/System.Net.Sockets/src/System.Net.Sockets.csproj

```diff
@@ -223,9 +213,178 @@
   </ItemGroup>
 
   <ItemGroup Condition=" '$(TargetsWindows)' == 'true' ">
-    <!-- TODO -->
+    <Compile Include="System\Net\Sockets\_AcceptOverlappedAsyncResult.Windows.cs" />
+    <Compile Include="System\Net\Sockets\_BaseOverlappedAsyncResult.Windows.cs" />
+    <Compile Include="System\Net\Sockets\_ConnectOverlappedAsyncResult.Windows.cs" />
+    <Compile Include="System\Net\Sockets\_OverlappedAsyncResult.Windows.cs" />
+    <Compile Include="System\Net\Sockets\Socket.Windows.cs" />
+    <Compile Include="System\Net\Sockets\SocketAsyncEventArgs.Windows.cs" />
```

Comment:
That's effectively what SocketAsyncEventArgs.{Windows,Unix}.cs are: all of the platform-specific implementation details for `SocketAsyncEventArgs` are within these files. The implementation details need to access enough of the internal state of the type that it felt more efficient to lay the PAL out this way (i.e. as part of the type itself).